src/config.rs

```diff
@@ -0,0 +1,90 @@
+use crate::NodeFormatter;
+
+/// The Colors that can be displayed in the console
+#[allow(missing_docs)]
+#[derive(Debug, PartialEq, Clone)]
+pub enum Color {
+    Black,
+    White,
+    Red,
+    Green,
+    Yellow,
+    Blue,
+    Magenta,
+    Cyan,
+    /// This allows for custom ANSI Colors to be specified, the specified Color Code will not be
+    /// valided and the User of this API needs to make sure that its a valid code
+    Custom(usize),
+}
+
+impl From<Color> for usize {
+    fn from(color: Color) -> Self {
+        match color {
+            Color::Black => 30,
+            Color::White => 37,
+            Color::Red => 31,
+            Color::Green => 32,
+            Color::Yellow => 33,
+            Color::Blue => 34,
+            Color::Magenta => 35,
+            Color::Cyan => 36,
+            Color::Custom(c) => c,
+        }
+    }
+}
+
+/// The Configuration to use for displaying a Graph
+///
+/// # Example
+/// ```rust
+/// use termgraph::{Config, IDFormatter};
+///
+/// let config: Config<usize, usize> = Config::new(IDFormatter::new(), 3).default_colors();
+/// ```
+pub struct Config<ID, T> {
+    pub(crate) formatter: Box<dyn NodeFormatter<ID, T>>,
+    pub(crate) color_palette: Option<Vec<Color>>,
+    pub(crate) max_per_layer: usize,
+}
+
+impl<ID, T> Config<ID, T> {
+    /// Creates a new Config with the given Formatter and Layer-Size
```

Comment:
Its unclear what layer-size means

Also, is there a sufficient default that to not require it in the constructor?

It'd be nice but not required if there was a default for the formatter as well from an API perspective.  Overall, boxing and throwing away the result if the user has a different formatting is probably not too big of a deal compared to the rest of the processing this crate does.  If there is concern about it, one option would be to make the field an `Option<T>` and have a getter that the rest of the code calls that does an `unwrap_or` with the default.

Reply:
Thanks for the Feedback, Layer-Size just means the maximum number of Nodes per Layer, im gonna change that in the Docs to make it more clear.
I would guess that you could probably have like 10 as a default because there is not too much horizontal space in your terminal, however as this is highly dependant on your application im not sure if its very useful if most people will just overwrite it.

For the second Part with the Formatter, Im not actually sure what the default should be, because displaying the ID or Value both make sense. The Part with the unnecessary boxing is also not really a concern right now, as there are plenty more cases in this crate that could be optimized, and I will get to them at some point. Right now it mainly comes back to not knowing what a good/sensible default would be